Framework/DataHandling/src/LoadInstrument.cpp

```diff
@@ -98,35 +97,36 @@ void LoadInstrument::init() {
  */
 void LoadInstrument::exec() {
   // Get the input workspace
-  m_workspace = getProperty("Workspace");
-  m_filename = getPropertyValue("Filename");
-  m_instName = getPropertyValue("InstrumentName");
+  boost::shared_ptr<API::MatrixWorkspace> ws = getProperty("Workspace");
+  std::string filename = getPropertyValue("Filename");
+  std::string instname = getPropertyValue("InstrumentName");
 
   // Decide whether to use Nexus or IDF loading
   // Note: for now, if only the instrument name is provided, the IDF loader will
   // be used by default. This should be updated in the future, so that the
   // Nexus loader can also take in an instrument name. Possibly by adding
   // a "FileType" property to the LoadInstrument algorithm.
-  if (LoadGeometry::isIDF(m_filename, m_instName))
-    idfInstrumentLoader();
-  else if (LoadGeometry::isNexus(m_filename))
-    nexusInstrumentLoader();
+  if (LoadGeometry::isIDF(filename, instname))
+    idfInstrumentLoader(ws, filename, instname);
+  else if (LoadGeometry::isNexus(filename))
+    nexusInstrumentLoader(ws, filename, instname);
   else
     throw Kernel::Exception::FileError("Instrument input cannot be read",
-                                       m_filename);
+                                       filename);
 
   // Set the monitors output property
-  setProperty("MonitorList", (m_workspace->getInstrument())->getMonitors());
+  setProperty("MonitorList", (ws->getInstrument())->getMonitors());
 
   // Rebuild the spectra map for this workspace so that it matches the
   // instrument, if required
   const OptionalBool RewriteSpectraMap = getProperty("RewriteSpectraMap");
   if (RewriteSpectraMap == OptionalBool::True)
-    m_workspace->rebuildSpectraMapping();
+    ws->rebuildSpectraMapping();
 }
 
 /// Load instrument from IDF XML file
-void LoadInstrument::idfInstrumentLoader() {
+void LoadInstrument::idfInstrumentLoader(boost::shared_ptr<API::MatrixWorkspace>& ws,
```

Comment:
Not sure it's worth the effort, but this is now also headed in the direction of being `const` or also a non-member. I don't suggest doing that as part of this PR though.

Reply:
so could I also make the `runLoadParameterFile()` and `getFullPathParamIDF()` non-member? They don't seem to be used anywhere else either

Reply:
i actually have a problem with making that one non-member because it's using `this` in the progress reporting: https://github.com/mantidproject/mantid/blob/23755_nexus_geom_in_LoadInstrument/Framework/DataHandling/src/LoadInstrument.cpp#L211